tests/unit/test_routes/test_searching.py

```diff
@@ -0,0 +1,212 @@
+
+    assert (response.status_code == 500)
+    assert ("Algolia" in response.get_json().get("errors")[0].get("algolia-failed").get(
+        "message"))
+
+    updated_term = random_string()
+
+    response = client.put("/api/v1/resources/1",
+                          json=dict(
+                              name="New name",
+                              languages=["New language"],
+                              category="New Category",
+                              url=f"https://{updated_term}.url",
+                              paid=False,
+                              notes="New notes"
+                          ),
+                          headers={'x-apikey': apikey}
+                          )
+
+    assert (response.status_code == 500)
+    assert ("Algolia" in response.get_json().get("errors")[0].get("algolia-failed").get(
+        "message"))
+
+    # Set it back to development for other tests
+    environ["FLASK_ENV"] = "development"
+
+
+def test_algolia_unreachable_host_error(module_client,
+                                        module_db,
+                                        fake_auth_from_oc,
+                                        fake_algolia_unreachable_host,):
+    # Reset the FLASK_ENV to run Algolia error handling code
+    environ["FLASK_ENV"] = "not_development"
+    client = module_client
+    first_term = random_string()
+    apikey = get_api_key(client)
+
+    response = client.get("/api/v1/search?q=python")
+
+    assert (response.status_code == 500)
+    assert ("Algolia" in response.get_json().get("errors")[0].get("algolia-failed").get(
+        "message"))
+
+    response = client.post("/api/v1/resources",
+                           json=[dict(
+                               name=f"{first_term}",
+                               category="Website",
+                               url=f"{first_term}",
+                               paid=False,
+                           )],
+                           headers={'x-apikey': apikey})
+
+    assert (response.status_code == 500)
+    assert ("Algolia" in response.get_json().get("errors")[0].get("algolia-failed").get(
+        "message"))
+
+    updated_term = random_string()
+
+    response = client.put("/api/v1/resources/1",
+                          json=dict(
+                              name="New name",
+                              languages=["New language"],
+                              category="New Category",
+                              url=f"https://{updated_term}.url",
+                              paid=False,
+                              notes="New notes"
+                          ),
+                          headers={'x-apikey': apikey}
+                          )
+
+    assert (response.status_code == 500)
+    assert ("Algolia" in response.get_json().get("errors")[0].get("algolia-failed").get(
+        "message"))
+
+    # Set it back to development for other tests
+    environ["FLASK_ENV"] = "development"
```

Comment:
Isn't this test the same as the previous test? Should we separate the body into a separate function and just make each test call it with different parameters? Or would that be overdoing it?

Reply:
Yes, it's the same test except the mocked exception is `fake_algolia_unreachable_host` instead of `fake_algolia_exception` (which is two different errors we want to ensure we're catching in these 4 situations).

If you want to tighten this up and combine them somehow or break the logic out into a function, feel free, just so long as it's still readable